devDungeon/src/systems/FallingSystem.java

```diff
@@ -82,12 +82,10 @@ private void teleportPlayerIfPossible() {
   }
 
   private Tile getSafeTile(Point heroCoords) {
-    Tile tile;
-    try {
-      tile = LevelUtils.accessibleTilesInRange(heroCoords, 5).getFirst();
-    } catch (NoSuchElementException e) {
-      tile = Game.randomTile(LevelElement.FLOOR);
-    }
-    return tile;
+    return LevelUtils.accessibleTilesInRange(heroCoords, 5).stream()
+        .findFirst()
+        .orElse(
+            Game.randomTile(LevelElement.FLOOR)
+                .orElseThrow(() -> new NoSuchElementException("No Floor Tile in the level.")));
```

Comment:
Designfrage: Ich habe hier (wenn ich nichts finden kann) zwei Alternativen: (a) direkt eine Exception zu werfen, oder (b) ein leeres `Optional` zurückzuliefern und den Aufrufer entscheiden zu lassen, wie darauf reagiert werden soll.

Ich kann den Finger nicht wirklich drauf legen, aber irgendwie erscheint mir (b) interessanter.

Reply:
das ist hier eine private Methode im Dev-Dungeon. Für mich ist das hier der Aufrufer.

Reply:
Hmmm. Der Aufrufer ist `FallingSystem#teleportPlayerIfPossible`, letztlich sollte/müsste dort reagiert werden. 

Hier in `getSafeTile` muss ja nur das "sichere Tile" berechnet werden. Das Handling ist irgendwie eine Frage des Standpunktes: 
- Gehe ich davon aus, dass es eigentlich irgendwie immer ein betretbares Tile im gewissen Umkreis um den Hero geben muss? Dann ist eine Exception in dieser Methode die richtige Modellierung - das Fehlen eines freien Feldes ist dann ein Fehler.
- Gehe ich davon aus, dass es möglicherweise keine betretbaren Tiles im Umkreis um den Hero geben kann und das auch ein gültiger Zustand ist, dann wäre die Rückgabe eines `Optional<Tile>` sinnvoller.

Wenn ich den Code betrachte und auch das Spiel so anschaue, scheint mir der erste Standpunkt der vermutlich sinnvollere zu sein.

Reply:
Damit hat sich diese Diskussion erledigt aus meiner Sicht. Irgendwie sollte aber im Aufrufer (`teleportPlayerIfPossible`) ein try/catch rein, der platzt aktuell nämlich einfach weg.

Reply:
Hmmmm ... 

@AMatutat Tatsächlich war im ursprünglichen Code bereits ein try/catch drin: Wenn das `getFirst()` fehlschlug, wurde ein zufälliges Tile genommen ... Damit war der Aufrufer "safe". 

Jetzt haben wir aber bei `randomTile` neu ein `Optional`, d.h. das kann dann auch wieder schief gehen. D.h. wir **müssen** eigentlich im Aufrufer (`teleportPlayerIfPossible`) ein geeignetes try/catch einbauen, sonst haben wir da eine möglich nicht gefangene neue Exception (neu nach dem Refactoring in diesem PR).

Reply:
Hmmm, Projekt mal löschen in IJ und neu anlegen? Zwischendrin unbedingt auch den .idea-Ordner rausnehmen :)

Reply:
negativ... ich versteh es auch nicht wirklich. Andere Java 21 Features gehen ja ohne Probleme. 

Ich hab jetzt mal auf "gut glück" gehackt.

Reply:
> Edit: UFF..... mein Fehler. Hab im letzten Eck von IntelliJ noch ne Einstellung gefunden. 

Erzähl mal mehr. Wo hast Du was gefunden und wie behoben? Das könnte ggf. ja auch andere betreffen, dann wäre es gut, die Lösung zu wissen :)

Reply:
Hab mal was gebastelt.

Reply:
?

Reply:
> > Edit: UFF..... mein Fehler. Hab im letzten Eck von IntelliJ noch ne Einstellung gefunden.
> 
> Erzähl mal mehr. Wo hast Du was gefunden und wie behoben? Das könnte ggf. ja auch andere betreffen, dann wäre es gut, die Lösung zu wissen :)

Wenn das Projekt geladen ist, kannst du unter File -> Project Structure -> Project die Projekt-SDK und ggf. das Language Level konfigurieren. Das sollte bekannt sein und war bei mir richtig eingerichtet.

Wenn du aber im IntelliJ-Hauptmenü (also ohne geladenes Projekt) wieder zu File -> Project Structure -> Project gehst, kannst du dort das globale SDK setzen. In meinem Fall stand dort SDK 17, und es hat die projektspezifische Konfiguration überschrieben. Wenn ich "No SDK" auswähle, wird die projektspezifische Version verwendet.

Jetzt frag mich nicht, warum und wieso Gradle das nicht egal ist.

Ich vermute, ein Zusammenspiel aus mehreren Problemen. Projekt neu einladen (inkl. gelöschtem .idea-Ordner) + die oben beschriebenen Konfigurationen haben das Problem für mich gelöst.

Reply:
Ah, verstehe. Ich glaube, das ist eher ein IntelliJ-Problem - das Ding versucht, möglichst clever zu sein, und schießt dabei (für meinen Geschmack zu) oft übers Ziel hinaus. Und irgendwie kriegen die das nicht hin, dass wenn Gradle-Konfig, dann auch nur Gradle-Konfig.